lib/sourcemapper.js

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/** @const */ var _ = require('lodash');
+/** @const */ var async = require('async');
+/** @const */ var fs = require('fs');
+/** @const */ var path = require('path');
+
+/** @const */ var sourceMap = require('source-map');
+
+/** @define {string} */ var MAP_EXT = '.map';
+
+function create(sourcemapPaths, callback) {
+  var mapper = new SourceMapper();
+  var callList = Array.prototype.slice.call(sourcemapPaths)
+                 .map(function(path) {
+                   return function(callback) {
+                     processSourcemap(mapper.infoMap_, path, callback);
+                   };
+                 });
+
+  async.parallel(callList, function(err) {
+    if (err){
+      return callback(
+        new Error('An error occurred while processing the sourcemap files' +
+          err));
+    }
+
+    callback(null, mapper);
+  });
+}
+
+/**
+ * @param {!Map} infoMap The map that maps input source files to
+ *  SourceMapConsumer objects that are used to calculate mapping information
+ * @param {string} mapPath The path to the sourcemap file to process.  The
+ *  path should be relative to the process's current working directory
+ * @private
+ */
+function processSourcemap(infoMap, mapPath, callback) {
+  // this handles the case when the path is undefined, null, or
+  // the empty string
+  if (!mapPath || !_.endsWith(mapPath, MAP_EXT)){
+    return callback(new Error('The path ' + mapPath +
```

Comment:
[Zalgo](http://blog.izs.me/post/59142742143/designing-apis-for-asynchrony). This callback is supposed to be async.